G/GDAL/build_tarballs.jl

```diff
@@ -91,6 +107,8 @@ dependencies = [
     Dependency("Libtiff_jll"; compat="4.3"),
     Dependency("libgeotiff_jll"; compat="100.700.100"),
     Dependency("LibCURL_jll"; compat="7.73"),
+    Dependency("NetCDF_jll"; compat="400.902.5"),
+    Dependency("HDF5_jll"),
```

Comment:
Since these packages aren't available on all platforms, I'd prefer to use them only when available, which would also clear the dozens of warnings about these packages not being available: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=22093&view=results.  Filtering them properly though is a bit tricky, but I don't have the time to do it now